### PR TITLE
Add golden master tests for istio-iptables.sh

### PR DESCRIPTION
**Pull request by phil9909**
`istio-iptables.sh` is currently only test with the integration tests.
With this PR we introduce golden master tests which should improve confidence when introducing new features and during refactoring.

### CODE REVIEW

**Comment by jkbschmid** [comment]
This PR is also a step towards solving https://github.com/istio/istio/issues/13967.

File: Makefile [hunk 1]
```diff
@@ -490,6 +490,10 @@ pilot-test: pilot-agent
 istioctl-test: istioctl
 	go test -p 1 ${T} ./istioctl/...
 
+.PHONY: iptables-test
+iptables-test:
```
Comment (howardjohn) [inline Makefile]: nit: we already have some tests for the iptables in common-test target (added a few weeks ago I think). maybe they can be consolidated into one target, either moving those tests here or these tests to the common-test

Comment (jkbschmid) [reply Makefile]: Fixed.

**Comment by howardjohn** [comment]
@phil9909 test seems to be failing.

```
FAIL	istio.io/./tests/scripts/istio-iptables-test.sh/mode_tproxy_and_wildcard_port	0.000s

3 test(s) failed:
  - mode_tproxy_and_ipv6
  - wildcard_include_ip_range
  - mode_tproxy_and_wildcard_port
Makefile:520: recipe for target 'common-test' failed
make[1]: *** [common-test] Error 3
...
```

It is a bummer prow doesn't understand the tests too. So at the top it shows FAILURE (0 failed). Not sure if we can do anything about that though

**Comment by sdake** [comment]
@phil9909 would you create an issue tracker for this issue and assign to me please so I can add to the distroless epic?

Cheers
-steve

**Comment by jkbschmid** [comment]
> @phil9909 would you create an issue tracker for this issue and assign to me please so I can add to the distroless epic?

As Phil is on vacation: https://github.com/istio/istio/issues/14604

**Comment by howardjohn** [comment]
I don't understand IP tables enough to know if the output is correct, but given iptables have been behaving like this for quite some time I'll assume it is correct :slightly_smiling_face:.

Thanks for writing these, this will be great for catching regressions

**Pull request merged**